### PR TITLE
 Merge bitcoin#9956: Reorganise qa directory 

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
63d66ba Move src/test/bitcoin-util-test.py to test/util/bitcoin-util-test.py (John Newbery)
5b0bff4 Rename --enable-extended-rpc-tests to --enable-extended-functional-tests (John Newbery)
a9bd622 Rename test/pull-tester/rpc-tests.py to test/functional/test_runner.py (John Newbery)
c28ee91 Rename rpc-tests directory to functional (John Newbery)
00902c4 Rename qa directory to test (John Newbery)

Tree-SHA512: ee7125c0c647d81590177beef2c8852c4ef76fdcf888096d9d4d360562a01d8d3b453345c3040487b2a043935bd1e7e80018f34462d6e02262bedbe23edcc576

### CODE REVIEW

**Comment by UdjinM6** [comment]
`No rule to make target 'test/util/data/txcreatemultisig3.hex'` - probably the same issue as https://github.com/dashpay/dash/pull/2763#discussion_r264814092

**Comment by PastaPastaPasta** [comment]
probably resolved

**Comment by UdjinM6** [comment]
`./ci/test_integrationtests.sh: line 20: ./qa/pull-tester/rpc-tests.py: No such file or directory`
https://travis-ci.org/dashpay/dash/jobs/532912529#L3446

**Comment by PastaPastaPasta** [comment]
Building w/o error, (except linux for some reason) 
Squashing

**Comment by UdjinM6** [comment]
Should also move a bunch of dash-specific tests https://github.com/dashpay/dash/tree/cc3915822d501ddfd109340fc4f482e494c589b1/qa/rpc-tests

File: test/README.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
+This directory contains integration tests that test bitcoind and its
```
Comment (UdjinM6) [inline test/README.md]: Should dashify this file

s/bitcoind/dashd/
s/bitcoin-qt/dash-qt/

File: test/README.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                        Write tested RPC commands into this directory
+```
+
+If you set the environment variable `PYTHON_DEBUG=1` you will get some debug
+output (example: `PYTHON_DEBUG=1 test/functional/test_runner.py wallet`).
+
+A 200-block -regtest blockchain and wallets for four nodes
+is created the first time a regression test is run and
+is stored in the cache/ directory. Each node has 25 mature
+blocks (25*50=1250 BTC) in its wallet.
```
Comment (UdjinM6) [inline test/README.md]: 25*500 = 12500 DASH

File: test/functional/autoix-mempool.py [hunk 3]
```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env python3
```
Comment (UdjinM6) [inline test/functional/autoix-mempool.py]: this file should be dropped, replaced by `autois-mempool.py`

File: test/README.md (hunk 1 above)
Comment (PastaPastaPasta) [reply test/README.md]: https://github.com/dashpay/dash/pull/2912/commits/2147d78c1b1271b63b1965a9e8b0f99e1c698dbb

File: test/README.md (hunk 2 above)
Comment (PastaPastaPasta) [reply test/README.md]: https://github.com/dashpay/dash/pull/2912/commits/2147d78c1b1271b63b1965a9e8b0f99e1c698dbb

File: test/README.md [hunk 4]
```diff
@@ -80,13 +80,13 @@ to recover with:
...
 rm -rf cache
-killall bitcoind
+killall dashd
 ```
 
 Util tests
 ----------
 
-Util tests can be run locally by running `test/util/bitcoin-util-test.py`. 
+Util tests can be run locally by running `test/util/dash-util-test.py`.
```
Comment (UdjinM6) [inline test/README.md]: shouldn't touch this line :)

Comment (PastaPastaPasta) [reply test/README.md]: https://github.com/dashpay/dash/pull/2912/commits/381efe77390e7b28d91ed043b0c6d4c2abdc314d

**Comment by PastaPastaPasta** [comment]
building locally, you can squash on merge

**Comment by PastaPastaPasta** [comment]
force pushed

**Comment by UdjinM6** [comment]
needs rebase

**Comment by PastaPastaPasta** [comment]
rebased

**Pull request merged**